src/panels/lovelace/cards/energy/hui-energy-gas-graph-card.ts

```diff
@@ -166,7 +166,13 @@ export class HuiEnergyGasGraphCard
           x: {
             type: "time",
             suggestedMin: start.getTime(),
-            suggestedMax: end.getTime(),
+            suggestedMax:
+              end.getTime() -
+              (dayDifference > 35
+                ? 3600000 * 24 * 30
+                : dayDifference > 2
+                ? 3600000 * 24
+                : 3600000),
```

Comment:
I think you may be removing slightly too much. E.g. for a week display, the start/end goes from Sunday 00:00:00 to Saturday 23:59:59

If you subtract a whole day from the end, then you're now ending on Friday 23:59:59. As a consequence you can see the tick label for Saturday is no longer visible:

![image](https://github.com/home-assistant/frontend/assets/32912880/32cef8ac-0988-4833-a4b3-eb1ffe55c26b)

Maybe use setDate(1), setHour(0), setMinute(0), setSecond(0) ? I think you want the end to fall at the same time as the final datapoint.

Reply:
I agree, using the date methods is also a more elegant solution.

> I think you want the end to fall at the same time as the final datapoint.

Yes. The suggested max (the end) should fall right on the start of the last bar within the chart.